neo/SmartContract/ApplicationEngine.cs

```diff
@@ -604,30 +604,36 @@ private bool PreStepInto(OpCode nextOpcode)
             return true;
         }
 
-        public static ApplicationEngine Run(byte[] script, IScriptContainer container = null, Block persisting_block = null, bool testMode = false)
+        public static ApplicationEngine Run(byte[] script, Snapshot snapshot,
+            IScriptContainer container = null, Block persistingBlock = null, bool testMode = false, Fixed8 extraGAS = default(Fixed8))
         {
-            using (Snapshot snapshot = Blockchain.Singleton.GetSnapshot())
+            snapshot.PersistingBlock = persistingBlock ?? new Block
```

Comment:
If `snapshot.PersistingBlock` is not `null` and `persistentBlock` is `null`, should we keep the original value instead of creating a new block?